account.rs/src/wasm/src/account.rs

```diff
@@ -0,0 +1,345 @@
+        let encryption_seed = format!("{}encryption", seed);
+        let blake_hash = blake2_string(encryption_seed, ACCOUNT_HASH_LENGTH);
+        let secret_key_bytes: [u8; ACCOUNT_HASH_LENGTH] = blake_hash.as_slice().try_into().unwrap();
+        let secret_key = SecretKey::from(secret_key_bytes);
+        secret_key.as_bytes().to_vec()
+    }
+
+    #[wasm_bindgen(js_name = createEncryptionPublicKey)]
+    pub fn create_encryption_public_key(encryption_private_key: Vec<u8>) -> Vec<u8> {
+        let secret_key_bytes: [u8; ACCOUNT_HASH_LENGTH] =
+            encryption_private_key.try_into().unwrap();
+        let secret_key = SecretKey::from(secret_key_bytes);
+        secret_key.public_key().as_bytes().to_vec()
+    }
+
+    #[wasm_bindgen(js_name = generateSecret)]
+    pub fn generate_secret(seed: &str, domain: &str) -> Vec<u8> {
+        let input = format!("{}{}", seed, domain);
+        blake2_string(input, ACCOUNT_HASH_LENGTH)
+    }
+
+    pub fn sign(&self, commitment: String, merkle_path: u32) -> Vec<u8> {
+        let inputs: Vec<Vec<u8>> = vec![
+            self.private_key.to_vec(),
+            commitment.into_bytes(),
+            merkle_path.to_be_bytes().to_vec(),
+        ];
+        poseidon_hash(inputs).unwrap()
+    }
+
+    #[wasm_bindgen(js_name = encryptAes)]
+    pub fn encrypt_aes(&self, message: Vec<u8>, iv16: Vec<u8>) -> Vec<u8> {
+        let key = Key::from_slice(&self.aes_secret);
+        let cipher = Aes256GcmSiv::new(key);
+        let nonce = Nonce::from_slice(&iv16);
+        let ciphertext = cipher
+            .encrypt(nonce, message.as_ref())
+            .expect("encryption failure!");
+
+        let mut result = iv16;
+        result.extend(ciphertext);
+        result
+    }
+
+    #[wasm_bindgen(js_name = decryptAes)]
+    pub fn decrypt_aes(&self, encrypted: Vec<u8>) -> Vec<u8> {
+        let iv16 = encrypted[0..16].to_vec();
+        let cipher_text = encrypted[16..].to_vec();
+        self.decrypt_aes_internal(cipher_text, iv16)
+    }
+
+    fn decrypt_aes_internal(&self, cipher_text: Vec<u8>, iv16: Vec<u8>) -> Vec<u8> {
+        let key = Key::from_slice(&self.aes_secret);
+        let cipher = Aes256GcmSiv::new(key);
+        let nonce = Nonce::from_slice(&iv16);
+        let decrypted_message = cipher
+            .decrypt(nonce, cipher_text.as_ref())
+            .expect("decryption failure!");
+        decrypted_message
+    }
+
+    #[wasm_bindgen(js_name = getUtxoPrefixViewingKey)]
+    pub fn get_utxo_prefix_viewing_key(&self, salt: &str) -> Vec<u8> {
+        if let Some(hashing_secret) = &self.hashing_secret {
+            let input = hashing_secret.to_vec();
+            let hashing_secret_string = String::from_utf8(input).unwrap();
+            Account::generate_secret(&hashing_secret_string, salt)
+        } else {
+            panic!("hashing_secret is None")
+        }
+    }
+
+    fn generate_utxo_prefix_hash(
+        &self,
+        merkle_tree_public_key: Vec<u8>,
```

Comment:
I know that the main reason why we are using vectors is to provide a JS-friendly API, but I would insist on using slices (sized, if possible) or arrays in the inner functions, not annotated with `#[wasm_bindgen]`.

Especially in this case, public keys have deterministic size, so it would be nice to use the Rust type system to enforce the correct length.

```suggestion
        merkle_tree_public_key: &[u8; KEY_LENGTH],
```